services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -91,13 +93,15 @@ public Response performLogin(AuthenticationRequest request) {
                 protocolBinding = JBossSAMLURIConstants.SAML_HTTP_POST_BINDING.get();
             }
 
+            String loginHint = request.getHttpRequest().getUri().getQueryParameters().getFirst(LOGIN_HINT_QUERY_PARAMETER);
             SAML2AuthnRequestBuilder authnRequestBuilder = new SAML2AuthnRequestBuilder()
```

Comment:
Is there some input validation / escaping required here?

Reply:
This value is sanitized in the `subject` method of `SAML2AuthnRequestBuilder`. It is also escaped using Stax when it is written in the SAML request. Do you think something else is required?